SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -216,3 +239,44 @@ void FSpatialLoadBalancingHandler::LogMigrationFailure(EActorMigrationResult Act
 		}
 	}
 }
+
+bool FSpatialLoadBalancingHandler::EvaluateRemoteMigrationComponent(const AActor* Owner, const AActor* Target,
+																	UAbstractLBStrategy* LBStrategy, VirtualWorkerId& WorkerId)
+{
+	if (NetDriver->LockingPolicy->IsLocked(Owner))
+	{
+		return false;
+	}
+
+	if (Target != nullptr)
+	{
+		UE_LOG(LogRemotePossessionComponent, Log, TEXT("Component->Target is:%s"), *Target->GetName());
+		VirtualWorkerId ActorAuthVirtualWorkerId = LBStrategy->WhoShouldHaveAuthority(*Owner);
```

Comment:
Careful, what the LoadBalancingStrategy tells you might not be the actual authority.
The assumption entering this method is that the local worker has authority over the Owner actor, so no need to try to retrieve it from somewhere else.
You could also check Target->HasAuthority to determine that no migration is needed.
Then, to retrieve the destination, you should be looking at the AuthorityIntent component, like it is done in EvaluateSingleActor (ln78 in my workspace, you likely moved it below in yours) in the case of a child actor joining it owner.

Reply:
Awesome, I didn't notice the AuthorityIntent.